src/humanloop/decorators/prompt.py

```diff
@@ -0,0 +1,112 @@
+    presence_penalty: Optional[float] = None,
+    frequency_penalty: Optional[float] = None,
+):
+    def decorator(func: Callable):
+        decorator.__hl_file_id = uuid.uuid4()
+
+        if temperature is not None:
+            if not 0 <= temperature < 1:
+                raise ValueError(f"{func.__name__}: Temperature parameter must be between 0 and 1")
+
+        if top_p is not None:
+            if not 0 <= top_p <= 1:
+                raise ValueError(f"{func.__name__}: Top-p parameter must be between 0 and 1")
+
+        if presence_penalty is not None:
+            if not -2 <= presence_penalty <= 2:
+                raise ValueError(f"{func.__name__}: Presence penalty parameter must be between -2 and 2")
+
+        if frequency_penalty is not None:
+            if not -2 <= frequency_penalty <= 2:
+                raise ValueError(f"{func.__name__}: Frequency penalty parameter must be between -2 and 2")
+
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            with get_tracer().start_as_current_span(str(uuid.uuid4())) as span:
+                trace_metadata = get_trace_context()
+
+                if trace_metadata:
+                    # We are in a Flow context
+                    write_to_opentelemetry_span(
+                        span=span,
+                        key=HL_TRACE_METADATA_KEY,
+                        value={**trace_metadata, "is_flow_log": False},
+                    )
+                    # Set current Prompt to act as parent for Logs nested underneath
+                    push_trace_context(
+                        {
+                            **trace_metadata,
+                            "trace_parent_id": span.get_span_context().span_id,
+                            "is_flow_log": False,
+                        },
+                    )
+
+                write_to_opentelemetry_span(
+                    span=span,
+                    key=HL_FILE_OT_KEY,
+                    value={
+                        "path": path if path else func.__name__,
+                        "prompt": {
+                            "template": template,
+                            "temperature": temperature,
+                            "top_p": top_p,
+                            "presence_penalty": presence_penalty,
+                            "frequency_penalty": frequency_penalty,
+                            "model": model,
+                            "endpoint": endpoint,
+                            "provider": provider,
+                            "max_tokens": max_tokens,
+                            "stop": stop,
+                        },
+                    },
+                )
+
+                try:
+                    output = func(*args, **kwargs)
+                except Exception as e:
+                    # TODO Some fails coming from here, they result in a fast end or duplicate
+                    # spans outputted to the Humanloop API
+                    print(e)
```

Comment:
Remove print statement. Should we have a logger defined? 

Also this TODO is a blocker. Where are the tests for this in the SDK?

Reply:
Should we catch the error in the first place? any intercept exception stems from developer written code, and I don't want it to fail silently

Reply:
We should be logging errors if there are unhandled errors from code that is being wrapped

Reply:
I am now logging the errors for the user and for the log output is set to None while error is the stringified exception